### PR TITLE
Prevent warning while TracingExtensionProvider loading

### PR DESCRIPTION
**Pull request by Shikugawa**
Signed-off-by: Rei Shimizu <rei@tetrate.io>

We faced warning message while loading TracingExtensionProvider config with existing tracing provider.

```
021-04-24T13:41:44.007222Z warn validation found invalid extension provider (can be ignored if the given extension provider is not used): invalid extension provider zippy-kin: unsupported provider: 
```

In this PR, we add some follow-ups to prevent this warning on the new telemetry API with provider service enhancement.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[x] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by Shikugawa** [comment]
cc @douglas-reid

File: pkg/config/validation/extensionprovider.go [hunk 1]
```diff
@@ -87,6 +99,48 @@ func validateExtensionProviderEnvoyExtAuthzGRPC(config *meshconfig.MeshConfig_Ex
...
+	}
+	if err := validateExtensionProviderService(config.Service); err != nil {
+		errs = appendErrors(errs, err)
+	}
+	if err := ValidatePort(int(config.Port)); err != nil {
+		errs = appendErrors(errs, fmt.Errorf("invalid service port: %v", err))
+	}
+	return
+}
+
```
Comment (douglas-reid) [inline pkg/config/validation/extensionprovider.go:143]: do we want to add validation for OpenCensus too?

Comment (Shikugawa) [reply pkg/config/validation/extensionprovider.go:143]: added

File: pkg/config/validation/extensionprovider.go [hunk 2]
```diff
@@ -106,6 +173,12 @@ func validateExtensionProvider(config *meshconfig.MeshConfig) (errs error) {
 			currentErrs = appendErrors(currentErrs, validateExtensionProviderEnvoyExtAuthzHTTP(provider.EnvoyExtAuthzHttp))
 		case *meshconfig.MeshConfig_ExtensionProvider_EnvoyExtAuthzGrpc:
 			currentErrs = appendErrors(currentErrs, validateExtensionProviderEnvoyExtAuthzGRPC(provider.EnvoyExtAuthzGrpc))
+		case *meshconfig.MeshConfig_ExtensionProvider_Zipkin:
+			currentErrs = appendErrors(currentErrs, validateExtensionProviderTracingZipkin(provider.Zipkin))
+		case *meshconfig.MeshConfig_ExtensionProvider_Lightstep:
+			currentErrs = appendErrors(currentErrs, validateExtensionProviderTracingLightStep(provider.Lightstep))
+		case *meshconfig.MeshConfig_ExtensionProvider_Datadog:
+			currentErrs = appendErrors(currentErrs, validateExtensionProviderTracingDatadog(provider.Datadog))
```
Comment (douglas-reid) [inline pkg/config/validation/extensionprovider.go:181]: add OpenCensus here too?

Comment (Shikugawa) [reply pkg/config/validation/extensionprovider.go:181]: added

**Review by douglas-reid — APPROVED** [review]

**Comment by Shikugawa** [comment]
@istio/wg-networking-maintainers PTAL

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**